tests/update_workflow_test.go

```diff
@@ -434,10 +435,10 @@ func (s *integrationSuite) TestUpdateWorkflow_ValidateMessages() {
 			_, _, err = poller.PollAndProcessWorkflowTaskWithAttemptAndRetryAndForceNewWorkflowTask(false, false, false, false, 1, 5, false, nil)
 			if len(tc.RespondWorkflowTaskError) > 0 {
 				// respond workflow task should return error
-				s.Error(err, "RespondWorkflowTaskCompleted should return an error contains`%v`", tc.RespondWorkflowTaskError)
-				s.Contains(err.Error(), tc.RespondWorkflowTaskError)
+				require.Error(t, err, "RespondWorkflowTaskCompleted should return an error contains`%v`", tc.RespondWorkflowTaskError)
+				require.Contains(t, err.Error(), tc.RespondWorkflowTaskError)
 			} else {
-				s.NoError(err)
+				require.NoError(t, err)
```

Comment:
This block here is the loop body of a table-driven test within a Suite. Using s.Error or s.Contains here is wrong because it fails the *parent* test rather than the *current* test as we're iterating through the testcase data.